.travis.yml

```diff
@@ -1,10 +1,35 @@
 language: java
-jdk:
-  - openjdk8
-  - oraclejdk8
-dist: trusty
 
-install: true
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
 
-script:
- - ./gradlew build
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+
+jobs:
+ include:
+    - stage: testandvalidate
```

Comment:
here stage 'test' which is default

Reply:
test and validate would better describe it right now
the gradle check tasks (as mentioned below) just run junit5 tests
the third one (for speed reasons) just runs the static code analysis